src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -120,9 +120,9 @@ public void IncomingComplete(int count, Exception error)
                 {
                     _awaitableError = error;
                 }
-
-                Complete();
             }
+
+            Complete();
```

Comment:
Thought it couldn't :( 

I'm having trouble maxing CPU and `IncomingData` and `ConsumingComplete` are taking a much larger proportion of time then I remember - but its been a while; which would suggest locks.

Related then in `ConsumingComplete` can the interlocked `_consumingState` and block return be moved out of the lock?